lib/elixir_bench_web/router.ex

```diff
@@ -38,7 +38,7 @@ defmodule ElixirBenchWeb.Router do
         assign(conn, :runner, runner)
 
       {:error, _reason} ->
-        send_resp(conn, :unauthorized, Jason.encode!(%{error: "unauthorized"}))
+        halt(conn)
```

Comment:
why the change from unauthorized to halt?

Reply:
this `send_resp` here causes response already sent error... By `halting` we just do not propagate to the controller and give back the error instantly. BasicAuth itself has the responsibility of sending the unauthorized response when the callback fails.